frontend-opal/src/thunks/searchResult.js

```diff
@@ -1,33 +1,113 @@
 import actions from 'actions/searchResult';
 
-const data = [
-  {
-    name: 'MS',
-    fullName: 'Morgan Stanley',
-    market: 'NYSE'
-  },
-  {
-    name: 'MSFT',
-    fullName: 'Microsoft Corp.',
-    market: 'NASDAQ'
-  },
-  {
-    name: 'FBMS',
-    fullName: 'First Branchshares Inc/MS',
-    market: 'NASDAQ'
+import { IEXCLOUD_PUBLIC_KEY } from 'react-native-dotenv';
+
+import symbolData from './symbolData';
+
+// const data = [
+//   {
+//     name: 'MS',
+//     fullName: 'Morgan Stanley',
+//     market: 'NYSE'
+//   },
+//   {
+//     name: 'MSFT',
+//     fullName: 'Microsoft Corp.',
+//     market: 'NASDAQ'
+//   },
+//   {
+//     name: 'FBMS',
+//     fullName: 'First Branchshares Inc/MS',
+//     market: 'NASDAQ'
+//   }
+// ];
+
+async function fetchData() {
+  const symbols = `https://cloud.iexapis.com/stable/ref-data/symbols?token=${IEXCLOUD_PUBLIC_KEY}`;
+  const res = await fetch(symbols);
+  const data = await res.json();
+  return data;
+}
+
+function getExtractNum(data, num) {
+  const len = data.second.length;
+  return len / num > 1 ? num : len;
+}
+
+async function extractTopNResult(dataSet, num) {
+  const res = [];
+  // searchRes.first: 100% matched result; secondRes.second: partially matched results
+  res.push(...dataSet.first);
+  if (dataSet.second.length > 0) {
+    const count = getExtractNum(dataSet, num);
+    res.push(...dataSet.second.slice(0, count - 1));
   }
-];
+  return res;
```

Comment:
```javascript
async function extractTopNResult({first, second}, num) {
  return [...first, ...second].slice(0, num);
}
```

Reply:
Wow, only 1 line!!!

I feel shame of my codes :rofl:

Reply:
yeah well, a little overcomplicated it :)